android/sdk/src/main/cpp/app/organicmaps/sdk/bookmarks/data/Bookmark.cpp

```diff
@@ -0,0 +1,143 @@
+#include "map/bookmark.hpp"
+#include "app/organicmaps/sdk/Framework.hpp"
+#include "app/organicmaps/sdk/bookmarks/data/Metadata.hpp"
+#include "app/organicmaps/sdk/core/jni_helper.hpp"
+
+jobject CreateBookmark(JNIEnv * env, place_page::Info const & info, jni::TScopedLocalObjectArrayRef const & jrawTypes,
+                       jni::TScopedLocalRef const & routingPointInfo)
+{
+  // clang-format off
+  static jmethodID const ctorId = jni::GetConstructorID(env, g_bookmarkClazz,
+    "("
+    "Lapp/organicmaps/sdk/bookmarks/data/FeatureId;"  // featureId
+    "J"                                               // categoryId
+    "J"                                               // bookmarkId
+    "Ljava/lang/String;"                              // title
+    "Ljava/lang/String;"                              // secondaryTitle
+    "Ljava/lang/String;"                              // subtitle
+    "Ljava/lang/String;"                              // address
+    "Lapp/organicmaps/sdk/routing/RoutePointInfo;"    // routePointInfo
+    "I"                                               // openingMode
+    "Ljava/lang/String;"                              // wikiArticle
+    "Ljava/lang/String;"                              // osmDescription
+    "[Ljava/lang/String;"                             // rawTypes
+    ")V"
+  );
+  // clang-format on
+
+  // clang-format off
+  jobject mapObject = env->NewObject(g_bookmarkClazz, ctorId,
+    CreateFeatureId(env, info.GetID()),
+    static_cast<jlong>(info.GetBookmarkCategoryId()),
+    static_cast<jlong>(info.GetBookmarkId()),
+    jni::ToJavaString(env, info.GetTitle()),
+    jni::ToJavaString(env, info.GetSecondaryTitle()),
+    jni::ToJavaStringWithSupplementalCharsFix(env, info.GetSubtitle()),
+    jni::ToJavaString(env, info.GetSecondarySubtitle()),
```

Comment:
```ToJavaStringWithSupplementalCharsFix```
Probably, it should be used for all _titles_?

Reply:
Should we use `ToJavaStringWithSupplementalCharsFix` everywhere then? What about wiki/osm descriptions?